README.md

```diff
@@ -1,322 +1,329 @@
+## Quick-Start
 
-You can also query with a `QueryRequest` that lets you directly use the wire protocol. The FQL X wire protocol allows for queries as strings with arguments that safely inject into the string, like parameterized queries in other database drivers. Here's an example:
+```javascript
+const { Client, fql } = fauna;
 
-```typescript
-const result = await client.query<string>({
-  query: '"Hello " + arg1',
-  arguments: { arg1: "World!" },
+// configure your client
+const client = new Client({
+  secret: YOUR_FAUNA_SECRET
 });
-console.log(result);
-// prints { data: "Hello World!", txn_ts: 1667174400000000 } // txn_ts will vary!
-```
-
-You can also do that using the `fql` function! Like this:
 
-```typescript
-const greetee = "World!";
-const result = await client.query<string>(fql`"Hello " + ${greetee}`);
-console.log(result);
-// prints { data: "Hello World!", txn_ts: 1667174400000000 } // txn_ts will vary!
+try {
+  // create a Collection
+  const collection_query = fql`Collection.create({ name: "Dogs" })`;
+  const collection_result = await client.query(collection_query);
+
+  // create a Document
+  const document_query = fql`
+    Dogs.create({ name: "Scout" }) {
+      id,
+      ts,
+      name
+    }
+  `;
+  const document_result = await client.query(document_query);
+} catch (error: any) {
+  if (error instanceof fauna.FaunaError) {
+    // handle errors
+  }
+}
 ```
 
-This `fql` based query is identical to the first example - under the hood the driver will convert the `fql` input into a parameterized query composed of a string and arguments.
+## Querying with FQL X
 
-## Composing FQL X queries
+This driver uses a template-based approach to composing queries and operations. The advantage of this design is that you can prototype and test your queries in the Fauna dashboard shell, and then cut-and-paste those queries as templates in your client, which are executed in Fauna via this driver. You can parameterize your query by adding placeholders to the template, and then pass a set of arguments to the query() method, or resolve the placeholders with string interpolation.
 
-The `fql` function allows you to compose FQL X from other FQL X snippets. It then automatically forms your input into a fully-formed and correct Fauna query.
+You can write a query in pure FQL X using the driver's `fql` tag template function. Each FQL X language driver exposes the raw FQL X language - there is no need to learn a new framework for each language.
 
-Let's look at an example:
-
-```typescript
-// yields valid FQL X - can be used as a query or as part
-// of a larger query.
-function collectionExists(collectionName: string) {
-  return fql`Collection.byName(${collectionName}) != null`;
-}
+Here's an example in pure FQL X:
 
-let collectionName = "ComposedQuery";
-// below, we compose a FQL X query that injects the FQL X
-// generated by collectionExists
+```javascript
```

Comment:
this example feels pretty heavyweight IMO.

Its not easy to read at a glance, and you have to grok that its actually building an anonymous function in _FQL_ itself to understand how it works.

I'd keep it simple here to show the basic power.

Complex examples like this are going to make it difficult to understand what's going on.

The previous example met that simplicity criteria IMO.

Reply:
This came out of wanting to use a diverse example for the composition query. There we have 3 different kinds of composition:

- pure fql snippet
- reusable projection (similar to GraphQL fragments)
- inserting data

The pure fql query is then the exact same as the composition query.

I started off with a different pure fql example and had this example under query composition to explain what the composed one was doing. How does that sound to you?

Reply:
Quickstart and configuration examples are pretty good demonstrations at a glance. I wrapped up these sections into a new "Advanced Example" section at the bottom.